scripts/dats_jsonld_annotator/annotator.py

```diff
@@ -0,0 +1,323 @@
+                    vv, _schema, specific_context, context_dir
+                )
+                annotation_list.append(_json_o)
+                specific_context.update(_local_context)
+            json_object[k] = annotation_list
+        # Add the key mapping to the context
+        if k not in context.keys():
+            logger.debug(f'{k} not in context: {pal.Path(schema["id"]).name}')
+            continue
+        if k not in specific_context.keys():
+            specific_context[k] = context[k]
+            logger.debug(f"{k} added to context: {context[k]}")
+        elif specific_context[k] == context[k]:
+            continue
+        else:
+            logger.debug(f"{k} duplicate: {specific_context[k]} vs {context[k]}")
+            continue
+
+    dtype = schema["properties"]["@type"]["enum"][0]
+    if dtype not in context.keys():
+        logger.debug(f'{dtype} not in context {schema["id"]}')
+    else:
+        specific_context[dtype] = context[dtype]
+    json_object["@type"] = dtype
+    return json_object, specific_context
+
+
+def gen_jsonld_outpath(dats_json_f, out_path):
+    """
+    This function generates a output file path for the annotated DATS jsonld file.
+
+    :param dats_json_f: the path to the original DATS instance file
+    :param out_path: the folder or new file path where the annotated DATS.jsonld file should be stored
+    :return:  the final DATS.jsonld file path
+    """
+    if not isinstance(dats_json_f, pal.Path):
+        dats_json_f = pal.Path(dats_json_f)
+    if out_path is not None and not isinstance(out_path, pal.Path):
+        out_path = pal.Path(out_path)
+
+    out_name = dats_json_f.with_suffix(".jsonld").name
+    if out_path is None:
+        # We save the output to the parent directory of the current DATS JSON file
+        out_path = dats_json_f.parent
+    elif out_path.suffix != "":
+        # We have most likely gotten a path to a non-existent file, let's use that
+        return out_path
+    elif not out_path == dats_json_f.parent:
+        # We are saving this somewhere other than the original folder
+        out_name = f"{dats_json_f.parent.name}_{out_name}"
+    if out_path.is_dir():
+        dats_jsonld_f = out_path / out_name
+    elif out_path.parent.is_dir():
+        # TODO: check if there is a way to explicitly look for a file that doesn't exist
+        logger.info(f"I will write to {out_path.resolve() = }")
+        dats_jsonld_f = out_path.parent / out_name
+    else:
+        raise Exception(
+            f"{out_path = } for {dats_json_f.resolve} is not a path to a file or directory. "
+            f"I don't know how to store the output"
+        )
+    return dats_jsonld_f
+
+
+def dats_to_jsonld(dats_f, schema_f, context_dir, out_path=None, clobber=False):
+    """
+    Helper function to load the inputs and store the annotated DATS.jsonld file
+    """
+    # TODO: log how many terms were not annotated because they are missing from context
```

Comment:
that would indeed be nice to have that summary once the script is done running!

Reply:
let's create an extension issue